include/ddc/for_each.hpp

```diff
@@ -33,6 +33,23 @@ void for_each_serial(
     }
 }
 
+template <class RetType, class Element, std::size_t N, class Functor, class... Is>
+KOKKOS_FUNCTION void annotated_for_each_serial(
+        std::array<Element, N> const& begin,
+        std::array<Element, N> const& end,
+        Functor const& f,
+        Is const&... is) noexcept
+{
+    static constexpr std::size_t I = sizeof...(Is);
+    if constexpr (I == N) {
+        f(RetType(is...));
+    } else {
+        for (Element ii = static_cast<int>(begin[I]); ii < end[I]; ++ii) {
```

Comment:
This `static_cast<int>` is a nonsense workaround since `Element` aliases `std::size_t`. Without it the loop does not iterate with `nvcc` in the new tests.

Reply:
What does it print if you remove the cast ?

Reply:
It prints `0` as expected 🤷

![image](https://github.com/user-attachments/assets/f36617ef-72cf-4c95-b543-1b84cab29837)

Reply:
(if I print something inside the loop it appears only once)

Reply:
If it can print something inside the loop it means it can iterate once ?

What is the value of end[I] ?

Have you tried to run in debug ?

Reply:
Yes sorry, by "it does not iterate" I meant it goes just once through the loop body.

`end[I]` is printed as `10`

I am exclusively working in Debug mode.

Reply:
```
Kokkos::printf("%i %i\n", end[I], end[I]);
Kokkos::printf("%i %i\n", static_cast<int>(end[I]), static_cast<int>(end[I]));
```

prints:

```
10 0
10 10
```

Like if accessing `begin[I]` and `end[I]` was buggy without the cast

Reply:
Here it is likely an incompatible format specifier passed to printf, it needs to match the type of integer you pass. You need to change %i if you want to print a size_t

Reply:
Ah right, changing it to %llu correctly prints 0 for begin[I] and 10 for end[I] in all situations so my last comment was irrelevant

Reply:
If you print from inside the loop with this correction, is it better ?

Reply:
sorry what do you want me to print inside the loop ?

Reply:
You could print `ii` ?

Reply:
`Kokkos::printf("%llu\n", ii);` prints `0`